### PR TITLE
purego: support float arguments in RegisterFunc

### PR DESCRIPTION
**Pull request by TotallyGamerJet**
This PR adds proper float support into RegisterFunc. It puts the first 8 in registers and the 9th onto the stack. It also checks beforehand that there aren't too many arguments.

### CODE REVIEW

File: func_darwin.go [hunk 1]
```diff
@@ -68,6 +68,40 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
...
 	}
+	if cfn == 0 {
+		panic("purego: cfn is nil")
+	}
+	{
+		// this code checks how many registers and stack this function will use
+		// to avoid crashing with too many arguments
+		var ints = 0
+		var floats = 0
+		var stack = 0
```
Comment (hajimehoshi) [inline func_darwin.go:79]:
```go
var ints int
var floats int
var stack int
```

File: func_darwin.go [hunk 2]
```diff
@@ -81,37 +115,69 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 				args = tmp
 			}
 		}
-		var sysargs = make([]uintptr, len(args))
+		var sysargs [maxArgs]uintptr
+		var stack = sysargs[numOfIntegerRegisters():]
+		var floats [8]float64
+		numInts := 0
+		numFloats := 0
+		numStack := 0
```
Comment (hajimehoshi) [inline func_darwin.go:123]:
```go
var numInts int
var numFloats int
var numStack int
```

**Review by hajimehoshi — APPROVED** [review]
LGTM

**Comment by hajimehoshi** [comment]
Will we have tests with float numbers later?

**Comment by TotallyGamerJet** [comment]
I do plan on having a test once I finish support for callbacks with float arguments!

**Pull request merged**